extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -76,6 +97,40 @@ private function getImageAsTitleObject( $imageName ) {
 		return $title;
 	}
 
+	private function getGalleryItems( $imageData ) {
+		preg_match( '/.(UNIQ.*QINU)./U', $imageData, $galleryMarkers );
+		$galleryImages = array();
+
+		foreach ( $galleryMarkers as $marker ) {
+			$galleryWikitext = PortableInfoboxDataBag::getInstance()->getGallery( $marker );
+			$galleryLines = preg_split('/\r\n|\n|\r/', $galleryWikitext);
+
+			foreach ( $galleryLines as $galleryImage) {
```

Comment:
I recommend incorporating feedback at this stage. The more code we can keep on the gallery extension parsing logic the better. Still, follow-up will be needed, but is there a rationale not to use the code that's already there?